### PR TITLE
Made dynamics limits a generic class, improved its function in TricycleDrive plugin

### PR DESCRIPTION
**Pull request by josephduchesne**
So Far:
- Made dynamics limits a generic class, improved its function in TricycleDrive plugin
- Added optional deceleration parameter
- Updated TricycleDrive plugin documentation.
- Added Dynamics Limits unit tests.
- Added linear dynamics to the existing angular dynamics limits for Tricycle drive
- Added dynamics limits and corresponding documentation to DiffDrive plugin

### CODE REVIEW

File: flatland_plugins/test/dynamics_limits_test.cpp [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+  EXPECT_NEAR(9.0, dynamics_limits.acceleration_limit_, 1e-3);
+  EXPECT_NEAR(9.0, dynamics_limits.deceleration_limit_, 1e-3);
+  EXPECT_NEAR(0.0, dynamics_limits.velocity_limit_, 1e-3);
+
+  double result;
+  result = dynamics_limits.Limit(0, 1.0, 0.05);  // Try to accelerate at 20m/s/s, when the limit is 9
+  EXPECT_NEAR(0.45, result, 1e-3);
+  result = dynamics_limits.Limit(1.0, 0.0, 0.05);  // Try to deccelerate at -20m/s/s, when the limit is 9
+  EXPECT_NEAR(0.55, result, 1e-3);
+  result = dynamics_limits.Limit(10.0, 0.0, 0.05);  // Try to deccelerate at -20m/s/s, when the limit is 9
```
Comment (avidbots-kenneth) [inline flatland_plugins/test/dynamics_limits_test.cpp:195]: Comment should say -200 m/s/s?

File: flatland_plugins/test/dynamics_limits_test.cpp [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+  dynamics_limits.Configure(config);
+
+  EXPECT_NEAR(1.0, dynamics_limits.acceleration_limit_, 1e-3);
+  EXPECT_NEAR(9.0, dynamics_limits.deceleration_limit_, 1e-3);
+  EXPECT_NEAR(0.0, dynamics_limits.velocity_limit_, 1e-3);
+
+  double result;
+  result = dynamics_limits.Limit(0, 1.0, 0.05);  // Try to accelerate at 20m/s/s, when the limit is 1 (acceleration)
+  EXPECT_NEAR(0.05, result, 1e-3);
+  result = dynamics_limits.Limit(0.97, 1.0, 0.05);  // Try to accelerate at <1m/s/s, when the limit is 1
```
Comment (avidbots-kenneth) [inline flatland_plugins/test/dynamics_limits_test.cpp:259]: What is the < sign for in the comment?

File: flatland_plugins/test/dynamics_limits_test.cpp (hunk 1 above)
Comment (josephduchesne) [reply flatland_plugins/test/dynamics_limits_test.cpp:195]: I'd thought this was 20m/s/s? We're trying to go from 1m/s to 0m/s in 0.05 seconds (0m/s-1m/s)/0.05s = 20m/s/s?

File: flatland_plugins/test/dynamics_limits_test.cpp (hunk 2 above)
Comment (josephduchesne) [reply flatland_plugins/test/dynamics_limits_test.cpp]: Fixed, not sure what I intended the < to mean here, so I removed it

File: flatland_plugins/test/dynamics_limits_test.cpp (hunk 1 above)
Comment (avidbots-kenneth) [reply flatland_plugins/test/dynamics_limits_test.cpp]: Some of the examples have a 10 m/s range, like this and the next 2, wasn't sure if that meant the deceleration was 10x of the previous example which is a 1 m/s range.

Comment (josephduchesne) [reply flatland_plugins/test/dynamics_limits_test.cpp]: oh, sorry, I was hung up on line 193, and missed 195. You're right!

**Pull request merged**